### PR TITLE
Connect liquidity

### LINKED ISSUE #9: Wire up "add liquidity" with the UI
Related to: near-clp/smart-contracts#64

### PR DESCRIPTION
**Pull request by amityadav0**
Closes #9

## Summary

Flow when calling from UI:
- User enters a token amount
- `calcNearAddLiquidity`: called and it will return NEAR price to show on UI
- `incAllowance`: called to increase the allowance for that token (when user clicks increase allowance button)
- `addLiquiduty`: called to add liquidity (when user clicks add liquidity button)

`calcNearAddLiquidity` :
 - Input: Takes tokenDetails with Input value of token from user.
 - returns `NEAR` required for adding liquidity:

- Update GitCDN link

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -20,7 +20,7 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
-    <script src="https://ghcdn.rawgit.org/robert-zaremba/near-clp/master/webapp/src/config.json"></script>
+    <script src="https://gitcdn.xyz/repo/near-clp/nearswap-ui/master/src/config.json"></script>
```
Comment (robert-zaremba) [inline src/index.html:23]: why using `gitcdn.xyz` rather than `ghcdn.rawgit.org`?

File: src/services/near-nep21-util.js [hunk 2]
```diff
@@ -320,44 +320,44 @@ export async function calcPriceFromOut(tokenProvide, tokenWant) {
...
 
-  const amountOUT = toYoctosString(tokenOUT.amount); //user input: I want 2 tokens out from the pool -> convert "2" to yoctos
+  const amountWant = toYoctosString(tokenWant.amount); //user input: I want 2 tokens out from the pool -> convert "2" to yoctos
 
-  const amountIN = tokenIN.amount; //computed amount he has to send into the pool, already in yoctos 
+  const amountProvide = tokenProvide.amount; //computed amount he has to send into the pool, already in yoctos 
 
-  if (tokenIN.type === "Native token") {
+  if (tokenProvide.type === "Native token") {
     // Native to NEP-21
```
Comment (robert-zaremba) [inline src/services/near-nep21-util.js:330]: we should add a check here, that `tokenWant.type === "NEP-21"` (and forbid native-native exchange).

Comment (robert-zaremba) [reply src/services/near-nep21-util.js:330]: Also, let's put the string values in a `const`.

File: src/services/near-nep21-util.js [hunk 3]
```diff
@@ -370,6 +370,18 @@ export async function swapFromOut(tokenIN, tokenOUT) {
   }
 }
 
+// returns NEAR amount needed for x amount of tokens while adding liquidity
+export async function calcNearAmt(tokenDetails) {
```
Comment (robert-zaremba) [inline src/services/near-nep21-util.js:374]: let's rename the function name, to not make a confusion with near amount for swaps.

How about: `calcNearAddLiquidity`?

**Comment by robert-zaremba** [comment]
The points in _Potential followups_ seam to be implemented, right? "Potential Followup" is to express what left and what we should do next related to changes in this PR.

File: src/services/near-nep21-util.js (hunk 2 above)
Comment (amityadav0) [reply src/services/near-nep21-util.js:330]: OK, adding same for NEP-21

File: src/services/near-nep21-util.js (hunk 3 above)
Comment (amityadav0) [reply src/services/near-nep21-util.js:374]: ok

File: src/index.html (hunk 1 above)
Comment (amityadav0) [reply src/index.html:23]: changing

**Comment by amityadav0** [comment]
yes, the functions are implemented, but they need to be called from UI but that is related UI PR. 
updating description

File: src/services/near-nep21-util.js (hunk 2 above)
Comment (robert-zaremba) [reply src/services/near-nep21-util.js:352]: No, the second case was OK. We support NEP-21 <-> NEP-21 swaps.

Comment (robert-zaremba) [reply src/services/near-nep21-util.js:352]: we can only check if in case of NEP21 swaps, the tokens are different.

Comment (amityadav0) [reply src/services/near-nep21-util.js:352]: Yes, i have updated only if:  `token addresses are same then the user can't do swap`

**Review by jamesondh — APPROVED** [review]

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**